lib/services/access_request.go

```diff
@@ -1307,19 +1329,22 @@ func (m *requestValidator) validate(ctx context.Context, req types.AccessRequest
 		// will await approval).
 		requestTTL, err := m.calculatePendingRequestTTL(req, now)
 		if err != nil {
-			return trace.Wrap(err)
+			return nil, trace.Wrap(err)
 		}
 		req.SetExpiry(now.Add(requestTTL))
 
 		if req.GetAssumeStartTime() != nil {
 			assumeStartTime := *req.GetAssumeStartTime()
 			if err := types.ValidateAssumeStartTime(assumeStartTime, accessExpiry, req.GetCreationTime()); err != nil {
-				return trace.Wrap(err)
+				return nil, trace.Wrap(err)
 			}
 		}
 	}
 
-	return nil
+	if m.opts.dryRun {
+		return enrichment, nil
+	}
+	return nil, nil
```

Comment:
This is quite scary interface. 

I would expect that the `err == nil` the object is valid 

Can we just req.SetDryRunEnrichment(enrichment) ?

Reply:
Done.